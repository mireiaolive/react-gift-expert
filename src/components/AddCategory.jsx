import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setInputValue] = useState(" ");

    //event.target.value
    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    };

    const onSubmit = (event) => {
        event.preventDefault();
        if (inputValue.trim().length <= 1) return;

        // setCategories((categories) => [inputValue, ...categories]);
        onNewCategory(inputValue.trim());
        setInputValue("");
    };

    return (
        <form onSubmit={(event) => onSubmit(event)} aria-label="form">
            <input
                type="text"
                placeholder="search gifs"
                value={inputValue}
                onChange={(event) => onInputChange(event)}
            />
        </form>
    );
};

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
};
