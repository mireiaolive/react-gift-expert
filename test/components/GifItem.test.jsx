import React from "react";
import { render, screen } from "@testing-library/react";
import { GifItem } from "./../../src/components/GifItem";

describe("Probando componente <GifItem/>", () => {
    const title = "Tim Burton Film GIF by Tech Noir";
    const url =
        "https://media4.giphy.com/media/l396BoOTIFem9xqQU/giphy.gif?cid=638e8c3cq4ztdf635cj607e70x2q3w7z0k6y8h04e3g6vw79&ep=v1_gifs_search&rid=giphy.gif&ct=g";

    test("debe hacer match con el snapshot", () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test("debe mostrar el url y el alt indicado de la imagen", () => {
        render(<GifItem title={title} url={url} />);
        const { src, alt } = screen.getByRole("img");
        expect(src).toBe(url);
        expect(alt).toBe(alt);
    });

    test("debe de mostrar el titulo en el componente", () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    });
});
