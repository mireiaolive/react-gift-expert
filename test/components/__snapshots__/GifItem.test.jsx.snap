// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando componente <GifItem/> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Tim Burton Film GIF by Tech Noir"
      src="https://media4.giphy.com/media/l396BoOTIFem9xqQU/giphy.gif?cid=638e8c3cq4ztdf635cj607e70x2q3w7z0k6y8h04e3g6vw79&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Tim Burton Film GIF by Tech Noir
    </p>
  </div>
</div>
`;
